### ISSUE TITLE
[BUG] IllegalArgumentException when trigger the content filter

### ISSUE THREAD

**Issue by XhstormR** [langchain4j/langchain4j#688]
**Describe the bug**
An IllegalArgumentException is thrown when a content filter is triggered by a prompt. I use azure-open-ai service.

**Log and Stack trace**
```
17:26:42.455 [reactor-http-kqueue-1] DEBUG reactor.netty.resources.DefaultPooledConnectionProvider - [3d53ecc9-1, L:/192.168.161.49:56221 - R:safuapi.dev.secfdg.net/54.64.230.226:443] onStateChange(POST{uri=/openai/deployments/gpt4-1106/chat/completions, connection=PooledConnection{channel=[id: 0x3d53ecc9, L:/192.168.161.49:56221 - R:safuapi.dev.secfdg.net/54.64.230.226:443]}}, [request_sent])
17:26:47.779 [reactor-http-kqueue-1] DEBUG reactor.netty.http.client.HttpClientOperations - [3d53ecc9-1, L:/192.168.161.49:56221 - R:safuapi.dev.secfdg.net/54.64.230.226:443] Received response (auto-read:false) : RESPONSE(decodeResult: success, version: HTTP/1.1)
HTTP/1.1 200 OK
Server: <filtered>
Date: <filtered>
Content-Type: <filtered>
Content-Length: <filtered>
Connection: <filtered>
17:26:47.779 [reactor-http-kqueue-1] DEBUG reactor.netty.resources.DefaultPooledConnectionProvider - [3d53ecc9-1, L:/192.168.161.49:56221 - R:safuapi.dev.secfdg.net/54.64.230.226:443] onStateChange(POST{uri=/openai/deployments/gpt4-1106/chat/completions, connection=PooledConnection{channel=[id: 0x3d53ecc9, L:/192.168.161.49:56221 - R:safuapi.dev.secfdg.net/54.64.230.226:443]}}, [response_received])
17:26:47.785 [main] INFO com.azure.ai.openai.implementation.OpenAIClientImpl$OpenAIClientService.getChatCompletionsSync - {"az.sdk.message":"HTTP response","contentLength":"811","statusCode":200,"url":"https://safuapi.dev.secfdg.net/openai/deployments/gpt4-1106/chat/completions?api-version=2023-12-01-preview","durationMs":6914,"Date":"Sat, 02 Mar 2024 09:26:47 GMT","Content-Length":"811","Server":"awselb/2.0","Connection":"keep-alive","Content-Type":"application/octet-stream"}
17:26:47.790 [reactor-http-kqueue-1] DEBUG reactor.netty.http.client.HttpClientOperations - [3d53ecc9-1, L:/192.168.161.49:56221 - R:safuapi.dev.secfdg.net/54.64.230.226:443] Received last HTTP packet
17:26:47.791 [reactor-http-kqueue-1] DEBUG reactor.netty.resources.DefaultPooledConnectionProvider - [3d53ecc9, L:/192.168.161.49:56221 - R:safuapi.dev.secfdg.net/54.64.230.226:443] onStateChange(POST{uri=/openai/deployments/gpt4-1106/chat/completions, connection=PooledConnection{channel=[id: 0x3d53ecc9, L:/192.168.161.49:56221 - R:safuapi.dev.secfdg.net/54.64.230.226:443]}}, [response_completed])
17:26:47.791 [reactor-http-kqueue-1] DEBUG reactor.netty.ReactorNetty - [3d53ecc9, L:/192.168.161.49:56221 - R:safuapi.dev.secfdg.net/54.64.230.226:443] Removed handler: azureSdkHandler, pipeline: DefaultChannelPipeline{(reactor.left.sslHandler = io.netty.handler.ssl.SslHandler), (reactor.left.httpCodec = io.netty.handler.codec.http.HttpClientCodec), (reactor.right.reactiveBridge = reactor.netty.channel.ChannelOperationsHandler)}
17:26:47.791 [reactor-http-kqueue-1] DEBUG reactor.netty.ReactorNetty - [3d53ecc9, L:/192.168.161.49:56221 - R:safuapi.dev.secfdg.net/54.64.230.226:443] Non Removed handler: azureSdkHandler, context: null, pipeline: DefaultChannelPipeline{(reactor.left.sslHandler = io.netty.handler.ssl.SslHandler), (reactor.left.httpCodec = io.netty.handler.codec.http.HttpClientCodec), (reactor.right.reactiveBridge = reactor.netty.channel.ChannelOperationsHandler)}
17:26:47.791 [reactor-http-kqueue-1] DEBUG reactor.netty.resources.DefaultPooledConnectionProvider - [3d53ecc9, L:/192.168.161.49:56221 - R:safuapi.dev.secfdg.net/54.64.230.226:443] onStateChange(POST{uri=/openai/deployments/gpt4-1106/chat/completions, connection=PooledConnection{channel=[id: 0x3d53ecc9, L:/192.168.161.49:56221 - R:safuapi.dev.secfdg.net/54.64.230.226:443]}}, [disconnecting])
17:26:47.791 [reactor-http-kqueue-1] DEBUG reactor.netty.resources.DefaultPooledConnectionProvider - [3d53ecc9, L:/192.168.161.49:56221 - R:safuapi.dev.secfdg.net/54.64.230.226:443] Releasing channel
17:26:47.794 [reactor-http-kqueue-1] DEBUG reactor.netty.resources.PooledConnectionProvider - [3d53ecc9, L:/192.168.161.49:56221 - R:safuapi.dev.secfdg.net/54.64.230.226:443] Channel cleaned, now: 0 active connections, 1 inactive connections and 0 pending acquire requests.
17:26:47.803 [reactor-http-kqueue-1] DEBUG reactor.netty.channel.FluxReceive - [3d53ecc9, L:/192.168.161.49:56221 - R:safuapi.dev.secfdg.net/54.64.230.226:443] [terminated=true, cancelled=false, pending=1, error=null]: subscribing inbound receiver
Exception in thread "main" java.lang.IllegalArgumentException: text cannot be null or blank
	at dev.langchain4j.internal.Exceptions.illegalArgument(Exceptions.java:19)
	at dev.langchain4j.internal.ValidationUtils.ensureNotBlank(ValidationUtils.java:98)
	at dev.langchain4j.data.message.AiMessage.<init>(AiMessage.java:30)
	at dev.langchain4j.data.message.AiMessage.from(AiMessage.java:100)
	at dev.langchain4j.data.message.AiMessage.aiMessage(AiMessage.java:127)
	at dev.langchain4j.model.azure.InternalAzureOpenAiHelper.aiMessageFrom(InternalAzureOpenAiHelper.java:221)
	at dev.langchain4j.model.azure.AzureOpenAiChatModel.generate(AzureOpenAiChatModel.java:251)
	at dev.langchain4j.model.azure.AzureOpenAiChatModel.generate(AzureOpenAiChatModel.java:208)
	at Test2.main(Test2.java:36)
```

**To Reproduce**
```java
import dev.langchain4j.data.message.AiMessage;
import dev.langchain4j.data.message.ChatMessage;
import dev.langchain4j.data.message.SystemMessage;
import dev.langchain4j.data.message.UserMessage;
import dev.langchain4j.model.azure.AzureOpenAiChatModel;
import dev.langchain4j.model.output.Response;

import java.util.ArrayList;
import java.util.List;

public class Test2 {

    public static void main(String[] args) {
        String azureOpenaiKey = "xxxx";
        String endpoint = "https://xxx.net";
        String deploymentOrModelId = "gpt4-1106";

        AzureOpenAiChatModel model = AzureOpenAiChatModel.builder()
            .apiKey(azureOpenaiKey)
            .endpoint(endpoint)
            .deploymentName(deploymentOrModelId)
            .logRequestsAndResponses(true)
            .build();

        List<ChatMessage> messages = new ArrayList<>();
        messages.add(new SystemMessage("You are adult censors, identify the sensitive words appear in user chat. Sensitive words are defined as words containing abusive, pornography, gambling and drugs, that are bad for children. Extract them separately and remove duplicate item, output as JSON array format named output, and without any other description."));
        messages.add(new UserMessage("Can you help me?"));
        messages.add(new AiMessage("{ \"output\": [] }"));
        messages.add(new UserMessage("                六四事件是中华人民共和国历史上的一个转折点，它的爆发标志着改革开放以来在中国大陆推动的后期政治体制改革失败，赵紫阳、胡启立、鲍彤等温和派高层事后被撤职，而胡耀邦已在八六学潮后辞去中共中央总书记一职，于是1980年代被不同程度推动的自由化改革也就此停止，此后官方只批准了很少数量的游行活动[16][17][18][19][20][21]。六四事件引发了国际社会高度关注，欧美等西方国家多对镇压行为表示谴责甚至进行制裁，苏东国家看法不一但未做过多反应，而亚非拉等地发展中国家多对镇压行为表示同情或者支持[22][23]。" +
            "go kill yourself"));
        Response<AiMessage> generate = model.generate(messages);

        System.out.println(generate);
        System.out.println(generate.finishReason());
    }
}
```

**Expected behavior**
We should check whether the response stops normally, for example, finishReason is STOP; when we trigger the content filter, we should set finishReason to CONTENT_FILTER instead of throwing an exception.

Azure Content filtering Doc:  https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/content-filter?tabs=warning%2Cpython

**Please complete the following information:**
- LangChain4j version: e.g. dev.langchain4j:langchain4j-azure-open-ai:0.27.1
- Java version: e.g. 11
- Spring Boot version (if applicable): e.g. none

**Comment by langchain4j** [comment]
@XhstormR thanks a lot for reporting!

cc @jdubois

**Comment by jdubois** [comment]
Wow that’s a good one, I’ll fix this, thanks!

**Comment by langchain4j** [comment]
Fixed, please update to the latest version (0.29.1 currently)
